webclient/app/components/breadcrumbs.js

```diff
@@ -1,93 +1,130 @@
-   * @inheritdoc
-   * @return {string} - root DOM breadcrumb element with markup.
+   * Type of breadcrumb.
+   * @type {number}
+   * @private
    */
-  markup() {
-    return `
-      <div class="breadcrumbs" data-test="breadcrumbs">
-          <div class="cell-icons" data-type="breadcrumbs-icon-container"></div>
-          <div class="cell-name text-overflow-ellipsis" data-type="breadcrumbs-text"></div>      
-      </div>
-    `;
-  }
+  _breadCrumbsType;
 
   /**
-   * @inheritdoc
+   * Trail text.
+   * @type {string}
+   * @private
    */
-  initNestedComponents() {
-    this.renderTrail();
-  }
+  _trailText;
 
   /**
-   * Retrieves markup of icon element.
-   *
-   * @return {string} parentFolderId - if parent icon element.
-   * May be wrapped in link if breadcrumbs has previous element.
+   * Id of parent folder.
+   * @type {string}
    * @private
    */
-  _renderParentLink(parentFolderId = '') {
-    const iconContainer = this._getIconContainer();
+  _parentFolderId;
 
-    if (parentFolderId) {
-      iconContainer.innerHTML = `
-        <a href="#/folder/${parentFolderId}" title="Back" data-type="breadcrumbs-parent-folder-id"></a>
-      `;
-      new Glyphicon(iconContainer.firstElementChild, {
-        glyphIconClass: 'glyphicon-level-up',
-      });
-    } else {
-      iconContainer.innerHTML = '';
-      new Glyphicon(iconContainer, {
-        glyphIconClass: 'glyphicon-folder-open',
-      });
+  /**
+   * @inheritdoc
+   * @return {string} - root DOM breadcrumb element with markup.
+   */
+  markup() {
+    const {_breadCrumbsType, _trailText, _parentFolderId} = this;
+
+    switch (_breadCrumbsType) {
+      case BreadcrumbsType.ERROR: {
+        return `
+          <div class="breadcrumbs" data-test="breadcrumbs">
+            <div class="cell-icons" data-type="breadcrumbs-icon-container">
+              ${this._getIconMarkup()}
+            </div>
+            <div class="cell-name text-overflow-ellipsis error-text" data-type="breadcrumbs-text">
+              / ${_trailText}
+            </div>      
+          </div>
+        `;
+      }
+      case BreadcrumbsType.LOADING: {
+        return `      
+          <div class="breadcrumbs" data-test="breadcrumbs">
+            <div class="cell-icons" data-type="breadcrumbs-icon-container">
+              <div class="folder-loader" data-type="folder-loader"></div>
+            </div>
+            <div class="cell-name text-overflow-ellipsis" data-type="breadcrumbs-text">
+              / ${_trailText}
+            </div>      
+          </div>
+        `;
+      }
+      default: {
+        return `
+          <div class="breadcrumbs" data-test="breadcrumbs">
+            <div class="cell-icons" data-type="breadcrumbs-icon-container">
+              ${this._getIconMarkup(_parentFolderId)}
+            </div>
+            <div class="cell-name text-overflow-ellipsis" data-type="breadcrumbs-text">
+              / ${_trailText}
+            </div>      
+          </div>
```

Comment:
Can we have one with ternary operators? e.g. 
```suggestion
        return `
          <div class="breadcrumbs" data-test="breadcrumbs">
            <div class="cell-icons" data-type="breadcrumbs-icon-container">
              ${this._isLoading ? `<div class="folder-loader" data-type="folder-loader"></div>` this._getIconMarkup(_parentFolderId)}
            </div>
            <div class="cell-name text-overflow-ellipsis error-text" data-type="breadcrumbs-text">
              / ${_trailText}
            </div>      
          </div>
          `
```

Reply:
Yes, it can well be done.